include/zephyr/sys/util.h

```diff
@@ -421,6 +421,21 @@ extern "C" {
  */
 #define IN_RANGE(val, min, max) ((val) >= (min) && (val) <= (max))
 
+/**
+ * @brief Map a value from one scale to another.
+ *
+ * A value of an input range is mappedto a value of an output range.
+ * The ranges are described by their min and max values.
+ *
+ * @param i The value to convert from the input range to the output range.
+ * @param imin The minimum value of the input range.
+ * @param imax The maximum value of the input range.
+ * @param omin The minimum value of the output range.
+ * @param omax The maximum value of the output range.
+ */
+#define MAP_VALUE(i, imin, imax, omin, omax) \
+	(((i) - (imin)) * (omin - omax) / ((imax) - (imin)) + (omin))
```

Comment:
Testing this using python:
```
def scale(i, imin, imax, omin, omax):
    return (((i) - (imin)) * (omin - omax) / ((imax) - (imin)) + (omin))

scale(2, 0, 10, 0, 100)
-20

scale(-2, 0, 10, 0, 100)
20

scale(-2, 0, 10, 0, 100)
20
```
sign is inverted :) Maybe should be:
```
(((i) - (imin)) * (omax - omin) / ((imax) - (imin)) + (omin))
```

Reply:
The tests could make sure the data is accurate, and that the overflows are known.
Thank you for your vigilance!